config/http.conf

```diff
@@ -9,15 +9,36 @@ http {
   client_max_body_size  400m;
   client_body_timeout   300s;
 
+  # If we receive X-Forwarded-Proto, pass it through; otherwise, pass along the
+  # scheme used to connect to this server
+  map $http_x_forwarded_proto $proxy_x_forwarded_proto {
+    default $http_x_forwarded_proto;
+    ''      $scheme;
+  }
```

Comment:
Basically this means, if the value of `$http_x_forwarded_proto` is `''` (empty or doesn't exist) then set a variable `$proxy_x_forwarded_proto` to `$scheme` ... and since that's the only other option in the `map` then if we don't match `''` then default to `default` which is defined as `$http_x_forwarded_proto` (itself) ... effectively passing it though if set.

Real world example: if using AWS LBs, `X-Forwarded-Proto` will be set by the LB to the protocol (`http` or `https`) that a client used to connect to your load balancer and therefore available as `$http_x_forwarded_proto`. See [classic-ELB](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/x-forwarded-headers.html#x-forwarded-proto) and [ALB](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/x-forwarded-headers.html#x-forwarded-proto) docs on the `X-Forwarded-Proto` header.